jetty-server/src/main/java/org/eclipse/jetty/server/HttpInput.java

```diff
@@ -139,9 +143,23 @@ public int read(byte[] b, int off, int len) throws IOException
     {
         synchronized (_inputQ)
         {
+            long now=System.currentTimeMillis();
+            
             if (_blockingTimeoutAt>=0 && !isAsync())
-                _blockingTimeoutAt=System.currentTimeMillis()+getHttpChannelState().getHttpChannel().getHttpConfiguration().getBlockingTimeout();
+                _blockingTimeoutAt=now+getHttpChannelState().getHttpChannel().getHttpConfiguration().getBlockingTimeout();
 
+            int minRequestDataRate=_channelState.getHttpChannel().getHttpConfiguration().getMinRequestDataRate();
+            if (minRequestDataRate>0)
+            {
+                long period=now-_channelState.getHttpChannel().getRequest().getTimeStamp();
+                if (period>=1000)
+                {
+                    double data_rate = _contentArrived / (0.001*(now-_channelState.getHttpChannel().getRequest().getTimeStamp()));
```

Comment:
Good points.  I'll add a field to hold the timestamp of the first byte to arrive.